File: Java/Principal_Exoneracion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package exoneracion;
import java.util.Scanner;
/**
 *
 * @author juanp
 */
public class Principal_Exoneracion {

    public static void main(String[] args) {
        // TODO code applicatio
        Scanner scanner = new Scanner(System.in);
        boolean datosCorrectos = false;
    do {    
    try {    
        System.out.print("Ingrese el nombre del visitante:\n ");
        String nombre = scanner.nextLine();
        System.out.print("Ingrese la edad del visitante:\n ");
        int edad = scanner.nextInt();
        scanner.nextLine();
        System.out.print("Ingrese la cédula del visitante:\n ");
        String cedula = scanner.nextLine();
        System.out.print("Ingrese el numero de carnet del visitante:\n ");
        String Numerocarnet = scanner.nextLine();
        
        visitante visitante1 = new visitante(nombre,edad,cedula,Numerocarnet);
        datosCorrectos = true;
    }catch(NumberFormatException e){
        System.out.println("Error: Ingrese un número válido para la edad.");
    }
    } while (!datosCorrectos);
    datosCorrectos = false;
        System.out.println("Se registro el visitante");
    do {    
    try {    
        System.out.print("Ingrese el nombre del cuidador:\n ");
        String nombre = scanner.nextLine();
        System.out.print("Ingrese la edad del cuidador:\n "); 
        int edad = scanner.nextInt();
        scanner.nextLine();
        System.out.print("Ingrese la cédula del cuidador:\n ");
        String cedula = scanner.nextLine();
        System.out.print("Ingrese el numero de carnet del cuidador:\n ");
        String Numerocarnet = scanner.nextLine();
        
        cuidador cuidador1 = new cuidador(nombre,edad,cedula,Numerocarnet);
        datosCorrectos = true;
    }catch(NumberFormatException e){
        System.out.println("Error: Ingrese un número válido para la edad.");
    }
    } while (!datosCorrectos); 
    datosCorrectos = false;
        System.out.println("se registro el cuidador");
     
    do {    
    try {    
        System.out.print("Ingrese el nombre del leon:\n ");
        String nombre = scanner.nextLine();
        System.out.print("Ingrese la edad del leon:\n ");
        int edad = scanner.nextInt();
        scanner.nextLine();
        System.out.print("Ingrese el Numero registro del leon:\n ");
        String Numeroregistro = scanner.nextLine();
        System.out.print("Ingrese el numero de carnet del leon:\n ");
        String vacuna = scanner.nextLine();
        
        Leon Leon1 = new Leon(nombre,edad,Numeroregistro,vacuna);
        datosCorrectos = true;
    }catch(NumberFormatException e){
        System.out.println("Error: Ingrese un número válido para la edad.");
    }
    } while (!datosCorrectos);
    datosCorrectos = false;
        System.out.println("se registro el leon");
        
    do {    
    try {    
        System.out.print("Ingrese el nombre del pinguino:\n ");
        String nombre = scanner.nextLine();
        System.out.print("Ingrese la edad: del pinguino:\n");
        int edad = scanner.nextInt();
        scanner.nextLine();
        System.out.print("Ingrese el numero registro: del pinguino:\n");
        String Numeroregistro = scanner.nextLine();
        System.out.print("Ingrese el numero de carnet:\n ");
        String Numerocarnet = scanner.nextLine();
        
        Pinguino Pinguino1 = new Pinguino (nombre,edad,Numeroregistro);
        datosCorrectos = true;
    }catch(NumberFormatException e){
        System.out.println("Error: Ingrese un número válido para la edad.");
    }
    } while (!datosCorrectos); 
        datosCorrectos = false;
    
        System.out.println("se registro el pinguino");
    }
}

File: Primer Parcial/C++/Principal_Exoneracion.cpp
#include <iostream>
#include <string>

class Visitante {
public:
    std::string Nombre;
    int Edad;
    std::string cedula;
    std::string Numerocarnet;

    Visitante(std::string nombre, int edad, std::string cedula, std::string Numerocarnet)
        : Nombre(nombre), Edad(edad), cedula(cedula), Numerocarnet(Numerocarnet) {}
};

class Cuidador {
public:
    std::string Nombre;
    int Edad;
    std::string cedula;
    std::string Numerocarnet;

    Cuidador(std::string nombre, int edad, std::string cedula, std::string Numerocarnet)
        : Nombre(nombre), Edad(edad), cedula(cedula), Numerocarnet(Numerocarnet) {}
};

class Leon {
public:
    std::string Nombre;
    int Edad;
    std::string Numeroregistro;
    std::string vacuna;

    Leon(std::string nombre, int edad, std::string Numeroregistro, std::string vacuna)
        : Nombre(nombre), Edad(edad), Numeroregistro(Numeroregistro), vacuna(vacuna) {}
};

class Pinguino {
public:
    std::string Nombre;
    int Edad;
    std::string Numeroregistro;
    std::string Numerocarnet;

    Pinguino(std::string nombre, int edad, std::string Numeroregistro, std::string Numerocarnet)
        : Nombre(nombre), Edad(edad), Numeroregistro(Numeroregistro), Numerocarnet(Numerocarnet) {}
};

int main() {
    std::string nombre, cedula, Numeroregistro, vacuna, Numerocarnet;
    int edad;

    bool datosCorrectos = false;
    do {
        try {
            std::cout << "Ingrese el nombre del visitante: " << std::endl;
            std::getline(std::cin, nombre);
            std::cout << "Ingrese la edad del visitante: " << std::endl;
            std::cin >> edad;
            std::cin.ignore();
            std::cout << "Ingrese la cédula del visitante: " << std::endl;
            std::getline(std::cin, cedula);
            std::cout << "Ingrese el número de carnet del visitante: " << std::endl;
            std::getline(std::cin, Numerocarnet);

            Visitante visitante1(nombre, edad, cedula, Numerocarnet);
            datosCorrectos = true;
        } catch (...) {
            std::cout << "Error: Ingrese un número válido para la edad." << std::endl;
        }
    } while (!datosCorrectos);

    datosCorrectos = false;
    std::cout << "Se registró el visitante" << std::endl;


    return 0;
}
